### ISSUE TITLE
(5)Unable to remove controlled values from subjectPlace when editing item metadata records

### ISSUE THREAD

**Issue by rmjaffe** [UCSCLibrary/ucsc-library-digital-collections#512]
### Describe the bug
In response to a correction submitted via LibAnswers I was wanting to update the location recorded in the metadata for following image, [Mariinsky Palace: two children](https://digitalcollections.library.ucsc.edu/records/t148fn067).  The record mistakenly indicated that this was the Catherine Palace in Pushkin; rather what is shown is the Mariinsky Palace in Saint Petersburg.  I was able to select and add Saint Petersburg as a value in subjectPlace but despite marking the values 'Catherine Palace' and 'Alexander Palace' for removal, they were not removed!

### To reproduce
**Steps to reproduce the behavior:**
1. Go to the image [Mariinsky Palace: two children](https://digitalcollections.library.ucsc.edu/records/t148fn067)
2. Click on Edit
3. Scroll down to subjectPlace and tick the 'x' for Catherine Palace and Alexander Palace

<img width="651" alt="Screen Shot 2022-11-16 at 3 37 23 PM" src="https://user-images.githubusercontent.com/32003407/202319142-5960a627-b793-4770-b085-12e42a6d0212.png">

4. Click save.
5. After you've saved and returned to the item show page, click 'more information' and surprise, the terms Catherine Palace and Alexander Palace will still show in the record.

![Screen Shot 2022-11-16 at 3 45 43 PM](https://user-images.githubusercontent.com/32003407/202319215-1639041c-1595-452c-a17a-cc662c88753f.png)

### Expected behavior
The terms I selected for removal should have been removed from the record and no longer display.

### Desktop (please complete the following information):
 - OS: Mac
 - Browser: Firefox

### Acceptance criteria
**Done means:**
- [x] Able to successfully remove controlled terms from subjectPlace
- [x] Values Catherine Palace and Alexander Palace have been removed from the metadata for this item

**Comment by rmjaffe** [comment]
Encountered the same issue in attempting to remove subjectPlace values 'Santa Cruz' and 'Eastside Santa Cruz' from the metadata for this image: https://digitalcollections.library.ucsc.edu/records/rf55z978f

**Comment by rmjaffe** [comment]
Just attempted to QA this in sandbox and ran into the same issue as before -- when I click on the edit button from one of the works, I land on the We’re sorry, but something went wrong page.

**Comment by snehagunduraoUL** [comment]
@rmjaffe could you please check what was the date format in the input csv? I see a date error again in the logs.

**Comment by rmjaffe** [comment]
I'm testing with this item: http://digitalcollections-staging-sandbox.library.ucsc.edu/records/8910jt584

The date value in the parent metadata record is 1979-06-15.  I'm now able to get to the edit form (yay!), but when I go to remove the subjectPlace value 'Upper Quarry Amphitheater' by clicking the x and saving the record, 'Upper Quarry Amphitheater' shows up again in the refreshed metadata.

**Comment by snehagunduraoUL** [comment]
I see. The removeFromList (from hyrax) functionality that we have used here removes subjectName values from a list. It is written generic to all the fields. If we have 3 subjectNames, we can remove 2 of them. We cannot empty the list. Similar to the title field.

**Comment by rmjaffe** [comment]
Ergh. It's always something!  Should I note this issue of not being able to empty lists in a new ticket?  We're going to need to be able to remove all values from a property for all the metadata properties that aren't required (i.e. everything except title).

**Comment by snehagunduraoUL** [comment]
Oh okay..I'll have to write some custom code for that. Sure, we can create a ticket for that. Do we need this change in production for now?

**Comment by rmjaffe** [comment]
Blocked by new ticket #550

**Comment by rmjaffe** [comment]
@snehagunduraoUL In terms of priority, this functionality is important (it would enable us to resolve some long outstanding patron metadata corrections) but it's not as urgent as the work that needs to be done in support of the aerials.  I'll put #550 in ready for development; should this remain in sandbox QA for the time being, or should I pull it back as well?

**Comment by snehagunduraoUL** [comment]
It can remain in sandbox QA. Should we push these changes only when we have 550 ready?

**Comment by rmjaffe** [comment]
OK, sorry -- I was so focused on that one example.  I added a couple more random values to subjectPlace and to subjectName and was able to successfully remove all values but one from each of those properties.  So, you're right.  This can move forward.

**Comment by rmjaffe** [comment]
Was able to remove subjectPlace values on the example image!  Will call this one done.
